server/StudySharp.ApplicationServices/Commands/RemoveTagByIdCommand.cs

```diff
@@ -13,18 +15,30 @@ public sealed class RemoveTagByIdCommand : IRequest<OperationResult>
         public int Id { get; set; }
     }
 
+    public class RemoveTagByIdCommandValidator : AbstractValidator<RemoveTagByIdCommand>
+    {
+        public RemoveTagByIdCommandValidator(ITagRules rules)
+        {
+            RuleFor(_ => _.Id)
+                .NotEmpty()
```

Comment:
Tag id will never be empty because it's not nullable. The default value for int is 0. So you can safely remove .NotEmpty() check for it